src/webapp/components/tag/ButtonTag.tsx

```diff
@@ -0,0 +1,75 @@
+import styled, { css } from "styled-components";
+import customTheme from "$/webapp/pages/app/themes/customTheme";
+import { memo } from "react";
+
+type Props = {
+    text: string;
+    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+export type ButtonProps = {
+    $text: Props["text"];
+};
+
+export const ButtonTag: React.FC<Props> = memo(({ text, onClick }) => {
+    return text ? (
+        <Button $text={text} onClick={onClick} aria-controls="simple-menu" aria-haspopup="true">
+            {text}
+        </Button>
+    ) : (
+        <Button $text={text} onClick={onClick} aria-controls="simple-menu" aria-haspopup="true">
```

Comment:
This condition should be handled in the parent component.

```tsx
// MyCustomComponent.tsx
<ButtonTag text={text ? text : "No Action"} onClick={onOpenMenu} />
```

Now we don't have a hardcoded value inside our component, making it more easy to reuse.